Helper/Cart.php

```diff
@@ -1374,6 +1370,93 @@ function ($item) use ($imageHelper, &$totalAmount, &$diff, $storeId, $currencyCo
         return [$products, $totalAmount, $diff];
     }
 
+    /**
+     * @param $item
+     * @return \Magento\Catalog\Model\Product
+     */
+    public function getProductToGetImageForQuoteItem($item)
+    {
+        $productType = $item->getProductType();
+        if ($productType == \Magento\ConfigurableProduct\Model\Product\Type\Configurable::TYPE_CODE) {
+            return $this->getProductToGetImageForConfigurableItem($item);
+        }
+
+        if ($productType == \Magento\GroupedProduct\Model\Product\Type\Grouped::TYPE_CODE) {
+            return $this->getProductToGetImageForGroupedItem($item);
+        }
+
+        return $item->getProduct();
+    }
+
+    /**
+     * @see \Magento\ConfigurableProduct\Block\Cart\Item\Renderer\Configurable::getProductForThumbnail logic
+     * (Magento version is less than 2.2.7)
+     *
+     * @see \Magento\ConfigurableProduct\Model\Product\Configuration\Item\ItemProductResolver::getProductForThumbnail logic
+     * (Magento version is greater or equal to 2.2.7)
+     *
+     * @param $item
+     * @return \Magento\Catalog\Model\Product
+     */
+    private function getProductToGetImageForConfigurableItem($item)
+    {
+        $parentProduct = $item->getProduct();
+        /** @var \Magento\Quote\Model\Quote\Item\Option $option */
+        $option = $item->getOptionByCode('simple_product');
+        $childProduct = $option ? $option->getProduct() : $item->getProduct();
```

Comment:
nit: you use $item->getProduct() although you have it in $parentProduct. The same in next function. Could I ask you to fix this in separate PR?

Reply:
Yes, I'll